dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/Serializer.java

```diff
@@ -0,0 +1,188 @@
+      } catch (MetricException me) {
+        logger.warning(String.format(TEMPLATE_ERR_METRIC_LINE, metric.getName(), me.getMessage()));
+      }
+    }
+    return lines;
+  }
+
+  @VisibleForTesting
+  List<String> createDoubleSumLines(MetricData metric, boolean isDelta) {
+    List<String> lines = new ArrayList<>();
+    for (DoublePointData point : metric.getDoubleSumData().getPoints()) {
+      try {
+        Metric.Builder builder = createMetricBuilder(metric, point);
+
+        if (isDelta) {
+          builder.setDoubleCounterValueDelta(point.getValue());
+        } else {
+          builder.setDoubleCounterValueTotal(point.getValue());
+        }
+
+        lines.add(builder.serialize());
+      } catch (MetricException me) {
+        logger.warning(String.format(TEMPLATE_ERR_METRIC_LINE, metric.getName(), me.getMessage()));
+      }
+    }
+    return lines;
+  }
+
+  @VisibleForTesting
+  List<String> createDoubleSummaryLines(MetricData metric) {
+    List<String> lines = new ArrayList<>();
+    for (DoubleSummaryPointData point : metric.getDoubleSummaryData().getPoints()) {
+      double min = Double.NaN;
+      double max = Double.NaN;
+      double sum = point.getSum();
+      long count = point.getCount();
+
+      List<ValueAtPercentile> percentileValues = point.getPercentileValues();
+      for (ValueAtPercentile percentileValue : percentileValues) {
+        if (Math.abs(percentileValue.getPercentile() - 0.0) < PERCENTILE_PRECISION) {
+          min = percentileValue.getValue();
+        } else if (Math.abs(percentileValue.getPercentile() - 100.0) < PERCENTILE_PRECISION) {
+          max = percentileValue.getValue();
+        }
+      }
+      if (Double.isNaN(min) || Double.isNaN(max)) {
+        logger.warning(
+            "The min and/or max value could not be retrieved. This happens if the 0% and 100% quantile are not set for the summary.");
+      }
+
+      try {
+        lines.add(
+            createMetricBuilder(metric, point)
+                .setDoubleSummaryValue(min, max, sum, count)
+                .serialize());
+      } catch (MetricException me) {
+        logger.warning(String.format(TEMPLATE_ERR_METRIC_LINE, metric.getName(), me.getMessage()));
+      }
+    }
+    return lines;
+  }
+
+  @VisibleForTesting
+  List<String> createDoubleHistogramLines(MetricData metric) {
+    List<String> lines = new ArrayList<>();
+    for (DoubleHistogramPointData point : metric.getDoubleHistogramData().getPoints()) {
+      double min = Double.NEGATIVE_INFINITY;
+      double max = Double.POSITIVE_INFINITY;
+      double sum = point.getSum();
+      long count = point.getCount();
+
+      List<Double> boundaries = point.getBoundaries();
+      // the histogram constructor checks that the boundaries are sorted in ascending order and
+      // that -Inf and Inf are not included.
+      if (boundaries.size() > 0) {
+        min = boundaries.get(0);
+        max = boundaries.get(boundaries.size() - 1);
```

Comment:
You'll need to walk through the buckets and take the first and last non-empty one - this will also add min/max for empty buckets. We can also do that in a follow-up PR, however.

Reply:
--> Follow-up story created.